src/App.js

```diff
@@ -1,34 +1,61 @@
-import React, { useState } from 'react';
-import { Route } from 'react-router-dom';
-import data from './data';
-
+import React, { useState } from "react";
+import { Route } from "react-router-dom";
+import data from "./data";
+import { ProductContext } from "./contexts/ProductContext";
+import { CartContext } from "./contexts/CartContext";
 // Components
-import Navigation from './components/Navigation';
-import Products from './components/Products';
-import ShoppingCart from './components/ShoppingCart';
+import Navigation from "./components/Navigation";
+import Products from "./components/Products";
+import ShoppingCart from "./components/ShoppingCart";
+
+const LOCALSTORAGE_KEY = "items";
 
 function App() {
-	const [products] = useState(data);
-	const [cart, setCart] = useState([]);
-
-	const addItem = item => {
-		// add the given item to the cart
-	};
-
-	return (
-		<div className="App">
-			<Navigation cart={cart} />
-
-			{/* Routes */}
-			<Route exact path="/">
-				<Products products={products} addItem={addItem} />
-			</Route>
-
-			<Route path="/cart">
-				<ShoppingCart cart={cart} />
-			</Route>
-		</div>
-	);
+  const [products] = useState(data);
+  const localStorageItems = window.localStorage.getItem(LOCALSTORAGE_KEY);
+  const [cart, setCart] = useState(
+    localStorageItems ? JSON.parse(localStorageItems) : []
+  );
```

Comment:
as sugera sa ii dai ca valoare initiala lui useState printr-o functie, daca tii minte noi asta numeai lazy state initialization

Reply:
adica in loc sa-i dau direct localStorageItems sa creez o functie care sa faca asta, si sa-i dau state-ului aceasta functie ?

Reply:
deci, acum la fiecare schimbare, la fiecare faza de "udpate" randul 15 se executa si asta nu e bine pentru performanta. Tu vrei de fapt linia 15 sa se execute doar atunci cand state-ul se initializeaza asai?

Reply:
iata un articol bun despre asta: https://kentcdodds.com/blog/use-state-lazy-initialization-and-function-updates

Reply:
Ok, am inteles ideea